File: src/ruby/test/test_name_validator.rb
require "minitest/autorun"
require_relative "../app/validators/name_validator.rb"

class TestNameValidator < Minitest::Test
  def test_validates_name
    test_data = [
      { :name => "Carl Mass", :expected => true },
      { :name => "Nicole Grownman", :expected => true },
      { :name => "Jean-Paul Satire", :expected => true },
      { :name => "Sophie Ballot Hawk", :expected => true },
      { :name => "john Doe", :expected => false },
      { :name => "John Do3", :expected => false },
      { :name => "John Doe.", :expected => false },
      { :name => "John", :expected => false },
      { :name => "John   Doe", :expected => false },
      { :name => "John-al Doe", :expected => false },
      { :name => "John DoE", :expected => false }
    ]

    test_data.each do |item|
      assert_equal(item[:expected], NameValidator.new(item[:name]).is_valid?, "#{item[:name]} validation failed")
    end
  end
end

File: src/python/validators/test_name_validator.py
import unittest
from name_validator import NameValidator

class TestNameValidator(unittest.TestCase):
    def test_validates_document(self):
        name_validator = NameValidator()

        test_data = [
            ("Carl Mass", True),
            ("Nicole Grownman", True),
            ("Jean-Paul Satire", True),
            ("Sophie Ballot Hawk", True),
            ("john Doe", False),
            ("John Do3", False),
            ("John Doe.", False),
            ("John", False),
            ("John   Doe", False),
            ("John-al Doe", False),
            ("John DoE", False)
        ]

        for item in test_data:
            self.assertEqual(item[1], name_validator.is_valid(item[0]))
